examples/robot_localization_failure/run.py

```diff
@@ -1,103 +1,311 @@
+                            pl.col("/scan/angle_increment").alias(
+                                "angle_increment",
+                            ),
+                            pl.col("/scan/range_min").alias("range_min"),
+                            pl.col("/scan/range_max").alias("range_max"),
+                        ],
+                    ).alias("/scan"),
+                    pl.col("/particle_cloud/particles").alias(
+                        "/particle_cloud",
+                    ),
+                    pl.struct(
+                        [
+                            pl.struct(
+                                [
+                                    pl.col(
+                                        "/amcl_pose/pose.pose.position.x",
+                                    ).alias("position.x"),
+                                    pl.col(
+                                        "/amcl_pose/pose.pose.position.y",
+                                    ).alias("position.y"),
+                                    pl.col(
+                                        "/amcl_pose/pose.pose.orientation.x",
+                                    ).alias("orientation.x"),
+                                    pl.col(
+                                        "/amcl_pose/pose.pose.orientation.y",
+                                    ).alias("orientation.y"),
+                                    pl.col(
+                                        "/amcl_pose/pose.pose.orientation.z",
+                                    ).alias("orientation.z"),
+                                    pl.col(
+                                        "/amcl_pose/pose.pose.orientation.w",
+                                    ).alias("orientation.w"),
+                                ],
+                            ).alias("pose"),
+                        ],
+                    ).alias("/amcl_pose"),
+                    pl.col("is_delocalized"),
+                )
+            )
+            # Split the data into train and test
+            data = DataFrame(data)
+            train, test = TrainTestSplit(ratio=0.8, shuffle=True).split(data)
+            train_images = FeatureImagesData(
+                train.data.collect(),
+                image_size=config.architecture.image_size,
+                width_meters=config.architecture.width_meters,
+            )
+            train_datasets.append(train_images)
+            test_images = FeatureImagesData(
+                test.data.collect(),
+                image_size=config.architecture.image_size,
+                width_meters=config.architecture.width_meters,
+            )
+            test_datasets.append(test_images)
+        except Exception as e:
+            msg = f"Error loading {parquet_path}: {e}"
+            logger.exception(msg)
+            continue
+
+    # Combinedatasets
+    train_data = pl.concat([train.data for train in train_datasets])
+    test_data = pl.concat([test.data for test in test_datasets])
```

Comment:
This sounds weird... you are first building a `FeatureImagesData` and then accessing the raw polars data again and rebuilding a new dataset using the factory below